File: LR_9-10/LR_10.pas
program SimpleMatrix;

const
  N = 8;

var
  a: array[1..N, 1..N] of integer;
  b: array[1..N] of integer;
  i, j: integer;
  found: boolean;

begin
  Randomize;
  
  for i := 1 to N do
    for j := 1 to N do
      a[i, j] := Random(10);
  
  writeln('Матрица:');
  for i := 1 to N do
  begin
    for j := 1 to N do
      write(a[i, j]:3);
    writeln;
  end;

  for i := 1 to N do
  begin
    found := false;
    for j := 1 to N - 1 do
      if a[i, j] + a[i, j + 1] = 7 then
        found := true;
    
    if found then
      b[i] := 1
    else
      b[i] := -1;
  end;

  writeln('Результат:');
  for i := 1 to N do
    write(b[i]:3);
end.

File: LR_9-10/LR_10.py
import random

N = 8


def main():
    # Инициализация матрицы
    a = [[0] * N for _ in range(N)]
    b = [0] * N

    # Заполнение матрицы случайными числами от 0 до 9
    random.seed()  # аналог Randomize в Pascal

    for i in range(N):
        for j in range(N):
            a[i][j] = random.randint(0, 9)

    # Вывод матрицы
    print('Матрица:')
    for i in range(N):
        for j in range(N):
            print(f'{a[i][j]:3}', end='')
        print()

    # Обработка строк матрицы
    for i in range(N):
        found = False
        # Проверяем пары соседних элементов в строке
        for j in range(N - 1):
            if a[i][j] + a[i][j + 1] == 7:
                found = True
                break  # можно прервать цикл, если нашли пару

        if found:
            b[i] = 1
        else:
            b[i] = -1

    # Вывод результата
    print('Результат:')
    for i in range(N):
        print(f'{b[i]:3}', end='')
    print()


if __name__ == "__main__":
    main()